src/frontend/src/models/Personnel.ts

```diff
@@ -2,6 +2,88 @@ export type PersonnelDiscipline = {
     name: string;
 };
 
+export type BasePosition = {
+    id: string;
+    name: string;
+    discipline: string;
+    projectType: string;
+};
+
+export type PositionContract = {
+    id: string;
+    name: string;
+    contractNumber: string;
+    company: {
+        id: string;
+        name: string;
+    };
+};
+
+export type PositionProject = {
+    projectId: string;
+    name: string;
+    domainId: string;
+    projectType: string;
+};
+
+export type Position = {
+    positionId: string;
+    instanceId: string;
+    name: string;
+    obs: string;
+    externalPositionId: string;
+    appliesFrom: Date;
+    appliesTo: Date;
+    workload: number;
+    basePosition: BasePosition;
+    project: PositionProject;
+    contract: PositionContract;
+};
+
+export type RequestPosition = {
+    id: string;
+    externalId: string;
+    name: string;
+    appliesFrom: Date;
+    appliesTo: Date;
+    workload: 0;
+    basePosition: BasePosition & {
+        wasResolved: true;
+    };
+    taskOwner: {
+        positionId: Date;
+    };
+};
+
+export type RequestContract = {
+    id: Date;
+    internalId: Date;
+    contractNumber: string;
+    name: string;
+    company: {
+        id: Date;
+        identifier: string;
+        name: string;
+    };
+};
+
+export type RequestProject = {
+    id: Date;
+    internalId: Date;
+    name: string;
+    projectMasterId: Date;
+};
+
+export type Requests = {
+    id: string;
+    created: Date;
+    updated: Date;
+    state: number;
+    position: RequestPosition;
+    contract: RequestContract;
+    project: RequestProject;
+};
+
```

Comment:
Don't we have almost every ones of these models already defined? Look at `PersonnelRequest.ts`

Reply:
Yes and no. 
They exist but there is always some discrepancy.
which could easily make it a mess if I tried to "shoehorn" those models in. 
 
Decided to just write out the entire model I found in swagger for the personnel endpoint. 
And put it into this one model.

Reply:
If the backend returns different models than `PersonnelRequest.ts` is mapped towards, I guess it is okay